legacy/application/models/StoredFile.php

```diff
@@ -394,16 +394,18 @@ public function delete($quiet = false)
             throw new DeleteScheduledFileException();
         }
 
-        $userInfo = Zend_Auth::getInstance()->getStorage()->read();
-        $user = new Application_Model_User($userInfo->id);
-        $isAdminOrPM = $user->isUserType([UTYPE_SUPERADMIN, UTYPE_ADMIN, UTYPE_PROGRAM_MANAGER]);
-        if (!$isAdminOrPM && $this->getFileOwnerId() != $user->getId()) {
-            throw new FileNoPermissionException();
+        // if we get here from the REST API, there's no valid user. APIKEY is validated already.
+        if ($userInfo = Zend_Auth::getInstance()->getStorage()->read()) {
+            // This call will throw "Trying to get property 'id' of non-object"
+            $user = new Application_Model_User($userInfo->id);
+            $isAdminOrPM = $user->isUserType([UTYPE_SUPERADMIN, UTYPE_ADMIN, UTYPE_PROGRAM_MANAGER]);
+            if (!$isAdminOrPM && $this->getFileOwnerId() != $user->getId()) {
+                throw new FileNoPermissionException();
+            }
+            $file_id = $this->_file->getDbId();
+            Logging::info($file_id);
+            Logging::info('User ' . $user->getLogin() . ' is deleting file: ' . $this->_file->getDbTrackTitle() . ' - file id: ' . $file_id);
```

Comment:
Shall we log a delete info message when deleted via API token, just without user information?